diff --git a/shared/dqn.py b/shared/dqn.py
index e02da3b..85bdc54 100644
--- a/shared/dqn.py
+++ b/shared/dqn.py
@@ -75,6 +75,8 @@ class DQN(Agent):
 
         self.q_list = list()
 
+        self._phi_list = list()
+
     def fit(self, dataset):
         s = np.array([d[0][0] for d in dataset]).ravel()
         games = np.unique(s)
@@ -120,6 +122,11 @@ class DQN(Agent):
             self._n_updates += 1
 
             if self._n_updates % self._target_update_frequency == 0:
+                game_state, game_action, game_reward, game_next_state,\
+                    game_absorbing, _ = self._replay_memory[0].get(
+                        self._batch_size)
+                _, phi = self.approximator.predict(game_state, idx=np.zeros(len(game_state), dtype=np.int), get_features=True)
+                self._phi_list.append(phi)
                 self._update_target()
 
     def get_shared_weights(self):
