diff --git a/shared/run_gym.py b/shared/run_gym.py
index a57bfcc..026cd7c 100644
--- a/shared/run_gym.py
+++ b/shared/run_gym.py
@@ -525,7 +525,7 @@ def experiment():
     if args.save_shared:
         pickle.dump(best_weights, open(args.save_shared, 'wb'))
 
-    return scores, losses, l1_losses, agent.q_list
+    return scores, losses, l1_losses, agent.q_list, agent._phi_list
 
 
 if __name__ == '__main__':
@@ -543,8 +543,10 @@ if __name__ == '__main__':
     loss = np.array([o[1] for o in out])
     l1_loss = np.array([o[2] for o in out])
     q = np.array([o[3] for o in out])
+    phis = np.array([o[4] for o in out])
 
     np.save(folder_name + '/scores.npy', scores)
     np.save(folder_name + '/loss.npy', loss)
     np.save(folder_name + '/l1_loss.npy', l1_loss)
     np.save(folder_name + '/q.npy', q)
+    np.save(folder_name + '/phis.npy', phis)
